src/Chapter3.hs

```diff
@@ -480,6 +513,42 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = Castle(String) | CastleWithWalls(String) | None deriving (Eq, Show)
+
+data Building = Church | Library deriving Show
+
+data PeopleAmount = One | Two | Three | Four deriving Show
+
+data House = House {
+  peopleAmount :: PeopleAmount
+} deriving Show
+
+data City = City {
+  castle :: Castle,
+  building :: Building,
+  houses :: [House]
+} deriving Show
+
+buildCastle :: City -> String -> City
+buildCastle city castleName = city{castle = Castle(castleName)}
```

Comment:
Constructors in Haskell are ordinary functions. They just start with an uppercase letter :top: 
So usual function calling conventions are applied here.

Also, I see that if the `Castle` already had walls, they will be removed by this function :bricks: :european_castle: :bricks: 
```suggestion
buildCastle city castleName = city{castle = Castle castleName}
```